pkg/kv/kvserver/multi_sst_writer.go

```diff
@@ -406,12 +403,8 @@ func (msstw *multiSSTWriter) PutRangeKey(
 
 	startKey, endKey := storage.EngineKey{Key: start}.Encode(), storage.EngineKey{Key: end}.Encode()
 	startTrailer := pebble.MakeInternalKeyTrailer(0, pebble.InternalKeyKindRangeKeySet)
-	msstw.rangeKeyFrag.Add(rangekey.Span{
-		Start: startKey,
-		End:   endKey,
-		Keys:  []rangekey.Key{{Trailer: startTrailer, Suffix: suffix, Value: value}},
-	})
-	return nil
+	rk := rangekey.Key{Trailer: startTrailer, Suffix: suffix, Value: value}
+	return msstw.PutInternalRangeKey(ctx, startKey, endKey, rk)
```

Comment:
There is some redundancy after this change: we encode the keys, and they are decoded back in `PutInternalRangeKey`. Also, `rolloverSST` is called twice - is this legit?

My preference would be to either denormalize this code again, or structure it so that there is no redundancy.